core/src/main/java/org/elasticsearch/search/aggregations/support/AggregationPath.java

```diff
@@ -295,21 +290,13 @@ public void validate(Aggregator root) {
 
                 // we're in the middle of the path, so the aggregator can only be a single-bucket aggregator
 
-                if (!(aggregator instanceof SingleBucketAggregator)) {
+                if (!(aggregator instanceof SingleBucketAggregator) && !(aggregator instanceof AggregatorFactory.MultiBucketAggregatorWrapper)) {
```

Comment:
The problem here is that now this check lets through any aggregation which implements `MultiBucketAggregatorWrapper` which is more than just the `NestedAggregator` (and `TermsAggregator`). So if I now try to sort be a histogram aggregation I get a very confusing error:
```
Invalid order path [histo_price>avg_price]. Metrics aggregations cannot have sub-aggregations (at [histo_price>avg_price]
```
Which is confusing because the histogram aggregation is not a metrics aggregation.

For the `NestedAggregator` case (and presumably `ReverseNestedAggregator` too) it might be a good idea to create a `SingleBucketAggregatorWrapper` class which does the same trick as the `MultiBucketAggregatorWrapper` but also implements `SingleBucketAggregator`. That way this check remains intact but we can support the nested aggregation in the terms ordering?